client/src/game/ui/HanabiCard.ts

```diff
@@ -939,6 +940,35 @@ export default class HanabiCard extends Konva.Group {
     this.setBareImage();
   }
 
+  // We need to redraw this card's suit and rank in a shared replay or hypothetical
+  // based on deckOrder and hypoRevealed
+  replayRedraw() {
+    if (globals.deckOrder.length === 0) {
+      return;
+    }
+    const suitNum = globals.deckOrder[this.order].suit;
+    const trueSuit = msgSuitToSuit(suitNum, globals.variant);
+    const trueRank = globals.deckOrder[this.order].rank;
+
+    if (
+      globals.hypothetical && !globals.hypoRevealed
```

Comment:
inb4 Zamiel commetns that you should have one condition per line

Reply:
now they are...